src/hooks/MessageComposer/useAttachments.ts

```diff
@@ -0,0 +1,41 @@
+import { useComposerStore } from "../../store/message-composer.store";
+import { prepareFileForUpload } from "../../service/upload-file-service";
+import { MAX_PAYLOAD_SIZE } from "../../config/constants";
+import { toast } from "../../utils/toast-helper";
+
+export const useAttachments = () => {
```

Comment:
proposal: rename `useAttachments` to `useAddAttachment` because this is the only exposed operation

general thought: we created an abstraction layer on top of `composerStore` and this layer should be the only entry-point for components wanting to add/process a file to an attachment. In this context, should we also consider renaming to `useAdaptFileToComposerAttachment` (or similar)?

This way we make it clear this hook is generic way of adding an attachment, while we still have the capability to set it "raw" via the store?